managed/services/backup/backup_service.go

```diff
@@ -505,3 +512,61 @@ func (s *Service) prepareBackupJob(
 
 	return res, dbConfig, nil
 }
+
+func (s *Service) checkArtifactModePreconditions(ctx context.Context, artifactID string, pitrTimestamp time.Time) error {
+	artifact, err := models.FindArtifactByID(s.db.Querier, artifactID)
+	if err != nil {
+		return err
+	}
+
+	if artifact.Vendor == string(models.MongoDBServiceType) && artifact.DataModel == models.PhysicalDataModel {
+		return errors.Wrapf(ErrIncompatibleService, "restore of physical backups is not supported for MongoDB yet")
+	}
+	if artifact.Vendor != string(models.MongoDBServiceType) && artifact.Mode == models.PITR {
+		return errors.Wrapf(ErrIncompatibleService, "restore to point in time is only available for MongoDB yet")
+	}
+
+	if artifact.Mode != models.PITR && pitrTimestamp.Unix() == 0 {
+		return nil
+	}
+	if artifact.Mode != models.PITR && pitrTimestamp.Unix() != 0 {
+		return errors.Wrapf(ErrIncompatibleArtifactMode, "artifact of type '%s' cannot be use to restore to point in time", artifact.Mode)
+	}
+	if artifact.Mode == models.PITR && pitrTimestamp.Unix() == 0 {
+		return errors.Wrapf(ErrIncompatibleArtifactMode, "artifact of type '%s' requires 'time' parameter to be restored to", artifact.Mode)
+	}
+	if artifact.Mode == models.PITR && artifact.DataModel == models.PhysicalDataModel {
+		return errors.Wrap(ErrIncompatibleArtifactMode, "point in time recovery is only available for Logical data model")
+	}
```

Comment:
Can we rewrite it like
```
if !pitr {
// pitr related checks
} else {
// simple restores checks
}
```

Otherwise it's hard to keep and track validation logic